client/layout/sidebar/expandable.jsx

```diff
@@ -56,12 +68,39 @@ export const ExpandableSidebarMenu = ( {
 	const classes = classNames( className, {
 		'is-toggle-open': !! expanded,
 		'is-togglable': true,
+		hovered: submenuHovered,
 	} );
 
+	const onEnter = () => {
+		if ( expanded || isTouch ) {
+			return;
+		}
+
+		setSubmenuHovered( true );
+	};
+
+	const onLeave = () => {
+		if ( expanded || isTouch ) {
+			return;
+		}
+
+		setSubmenuHovered( false );
```

Comment:
This will trigger a `setState` for `submenuHovered` which will then trigger the `useEffect` which will then trigger _another_ `setState` for `subMenuStyles`. Just wondering whether you could store the styles in a `ref` rather than state to avoid the double `setState` call. 

React might batch the `setState` calls but I'm not 100% sure given that the 2nd is in ` useEffect` so it's possible this will trigger x2 re-renders.

Reply:
Very good idea! Refactored 5bdb744